### DISCUSSION TITLE
How to implement B2 method in mrgsolve ?

### DISCUSSION THREAD

**Discussion by Ravua1992** [metrumresearchgroup/mrgsolve#1114] [Q&A]
Hello, 

B2 method is inclusion of the observed baseline response as a covariate acknowledging the residual variability> when I tired to implement this in mrgsolve I simulated negative values. 
Are there any good examples how to do this ? 

This is how I am doing it currently: 

mrgsolve in $MAIN: 
`double RTOT1 = RTOT1_OBS * (1.0 + ETA(15) * CCVRESi) + ETA(16) * ADDRESi;`

nonmem 

`  RTOT1  = RTOT1_OBS*(1.0 + ETA(15)*CCVRES) + ETA(16)*ADDRES`

Thanks,

**Comment by kylebaron** [comment]
Hi @Ravua1992  - 

I'm assuming that, when ETAs are 0, `RTOT1` comes out to be non-negative, right? 

If that's the case, and simulating ETA results in (some) negative `RTOT1`, then you can resimulate ETA until `RTOT1` is positive. 

The code might look like:

```
double RTOT1 = RTOT1_OBS * (1.0 + ETA(15) * CCVRESi) + ETA(16) * ADDRESi;

while(RTOT1 <= 0) {
  simeta();
  RTOT1 = RTOT1_OBS * (1.0 + ETA(15) * CCVRESi) + ETA(16) * ADDRESi;
}
```

You will have to make sure your code is structured so that the most-recently simulated set of ETA are used for the rest of the ETAs. 

Kyle

**Reply by curtisKJ** [reply]
Hey @Ravua1992 

Just a note, I think you need to be careful to realize that NONMEM is allowing for negative predicted/simulated values. To then set up the mrgsolve model in a way that precludes negative values will induce bias at the near zero values and won't be consistent with what NONMEM would simulate/predict if you used that instead.
